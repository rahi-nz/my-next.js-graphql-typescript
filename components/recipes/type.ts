export interface Props {
  photo: string;
  title: string;
  id: string;
}
