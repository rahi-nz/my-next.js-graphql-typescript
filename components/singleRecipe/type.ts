import { gql } from "apollo-boost";

export const RECIPE = gql`
  query($id: String!) {
    recipe(id: $id) {
      chef {
        name
      }
      description
      title
      photo {
        url
      }
      tagsCollection {
        items {
          name
        }
      }
    }
  }
`;

export interface Props {
  id: string;
}

interface chef {
  name: string | null;
}

interface photo {
  url: string;
}

interface tag {
  name: string;
}

interface tagsCollection {
  items: Array<tag>;
}

interface recipe {
  chef: chef;
  description: string;
  title: string;
  photo: photo;
  tagsCollection: tagsCollection;
}

export interface getRecipes {
  recipe: recipe;
}
