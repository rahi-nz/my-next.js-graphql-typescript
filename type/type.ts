import { gql } from "apollo-boost";

export const RECIPES = gql`
  query {
    recipeCollection {
      items {
        sys {
          id
        }
        title
        photo {
          url
        }
      }
    }
  }
`;

interface items {
  id: string;
}

interface photo {
  url: string;
}

interface items {
  sys: items;
  title: string;
  photo: photo;
}

interface Collection {
  items: Array<items>;
}

export interface recipesData {
  recipeCollection: Collection;
}
