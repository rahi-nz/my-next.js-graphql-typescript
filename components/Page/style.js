import styled from "styled-components";

export const Container = styled.section`
  text-align: center;
  width: 80%;
  margin: auto;
  padding: 20px;
`;

export const Element = styled.section`
  text-align: center;
  display: flex;
  flex-wrap: wrap;
  margin: auto;
  padding: 20px;
`;
