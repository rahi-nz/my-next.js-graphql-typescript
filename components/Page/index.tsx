import Head from "next/head";
import React, { ReactNode } from "react";
import { Container, Element } from "./style";

interface PropTypes {
  children: ReactNode;
  title: string;
  header: string;
}

export default function Page({ children, title, header }: PropTypes) {
  return (
    <Container>
      <Head>
        <title>{title}</title>
        <link rel="shortcut icon" href="/favicon/favicon.ico" />
        <link rel="shortcut icon" href="/favicon/favicon-16x16.png" />
        <link rel="shortcut icon" href="/favicon/favicon-32x32.png" />
        <link rel="shortcut icon" href="/favicon/favicon-96x96.png" />
      </Head>
      <h1>{header}</h1>
      <Element>{children}</Element>
    </Container>
  );
}
