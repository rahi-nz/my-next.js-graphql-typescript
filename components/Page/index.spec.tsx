import React from "react";
import renderer from "react-test-renderer";
import Page from "./index";
import { Props } from "./type";

describe("<Page/>", () => {
  let rootElement;
  let testRenderer;
  const prop: Props = {
    children: <p>children</p>,
    title: "title",
    header: "header",
  };
  // eslint-disable-next-line prefer-const,react/display-name
  rootElement = () => <Page {...prop} />;
  // eslint-disable-next-line prefer-const
  testRenderer = renderer.create(rootElement());

  it("Should render correctly", () => {
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
