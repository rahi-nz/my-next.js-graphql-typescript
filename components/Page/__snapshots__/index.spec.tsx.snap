// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page/> Should render correctly 1`] = `
<section
  className="style__Container-sc-16kkt56-0 bMKdLe"
>
  <h1>
    header
  </h1>
  <section
    className="style__Element-sc-16kkt56-1 beiIGD"
  >
    <p>
      children
    </p>
  </section>
</section>
`;
